sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -254,4 +254,13 @@ private[hive] trait HiveStrategies {
       case _ => Nil
     }
   }
+
+  object HiveDataSourceStrategy extends Strategy {
```

Comment:
Seems we do not need it. When we want to insert into a data source table, `logical.InsertIntoTable` will be used instead of `logical.InsertIntoHiveTable`.